internal/controllers/machinedeployment/mdutil/util.go

```diff
@@ -430,8 +430,6 @@ func FindNewMachineSet(deployment *clusterv1.MachineDeployment, msList []*cluste
 	var matchingMachineSets []*clusterv1.MachineSet
 	var diffs []string
 	for _, ms := range msList {
-		ms := ms
-
```

Comment:
I don't believe we will backport this but if we do, this may need to be there if 1.7.x and 1.6.x use anything less than go1.22.

Reply:
This was done to pass the linting:
` internal/controllers/machinedeployment/mdutil/util.go:433:3: The copy of the 'for' variable "ms" can be deleted (Go 1.22+) (copyloopvar)`

Reply:
If there is a need to backport this, I can do this off the branch where it will not impact the linting rules at that branch go version (e.g. go1.21)

Reply:
I'm curious why this comes up at this (unrelated to this file) pull request 🤔 

It should not be part of this PR.

Reply:
I'll open a PR to fix that separately.

Reply:
xref: #10731

Reply:
My concern with the removal of this to fix the linting, we will not have that loopvar in go1.21 & go1.20 (on the backports 1.6 & 1.7) where it may be necessary to have that within the scope of the loop (which 1.22 fixes).   

Even rebasing with the xref #10731 will still carry that in to the older releases which may impact how that works.  @chrischdi

Reply:
With merging #10731 and rebasing, this PR won't have the change to internal/controllers/machinedeployment/mdutil/util.go anymore and the cherry-picks should just go well. If there's still something missing then, then we have to do a manual cherry-pick I think.

Reply:
This has been rebased with main so I don't carry the linting change to the backports.  Thanks @chrischdi